axon-mongo-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/mongo/springboot/autoconfig/MongoAutoConfiguration.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.mongodb.MongoDatabaseFactory;
+import org.springframework.data.mongodb.MongoTransactionManager;
+
+import static java.util.Objects.isNull;
+
+@Configuration
+@EnableConfigurationProperties(AxonMongoProperties.class)
+@AutoConfigureAfter(name = {
+        "org.axonframework.springboot.autoconfig.TransactionAutoConfiguration",
+        "org.axonframework.springboot.autoconfig.JdbcAutoConfiguration"
+})
+public class MongoAutoConfiguration {
+
+    private final AxonMongoProperties axonMongoProperties;
+
+    public MongoAutoConfiguration(AxonMongoProperties axonMongoProperties) {
+        this.axonMongoProperties = axonMongoProperties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public MongoTransactionManager mongoTransactionManager(MongoDatabaseFactory factory) {
+        return new MongoTransactionManager(
+                factory,
+                TransactionOptions.builder().writeConcern(WriteConcern.MAJORITY).build()
+        );
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public MongoTemplate axonMongoTemplate(MongoDatabaseFactory factory) {
+        String databaseName = axonMongoProperties.getDatabaseName();
+        if (isNull(databaseName)) {
+            return SpringMongoTemplate.builder()
+                                      .factory(factory)
+                                      .build();
+        } else {
+            return SpringMongoTemplate.builder()
+                                      .factory(factory)
+                                      .databaseName(databaseName)
+                                      .build();
+        }
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public TransactionManager transactionManager(MongoTransactionManager manager) {
+        return new SpringMongoTransactionManager(manager);
+    }
+
+    @Bean("tokenStore")
+    @ConditionalOnMissingBean(TokenStore.class)
+    @ConditionalOnProperty(value = "axon.mongo.token-store.enabled", matchIfMissing = true)
+    public TokenStore tokenStore(
+            MongoTemplate mongoTemplate,
+            TransactionManager transactionManager,
+            Serializer serializer
+    ) {
+        return MongoTokenStore.builder()
+                              .mongoTemplate(mongoTemplate)
+                              .transactionManager(transactionManager)
+                              .serializer(serializer)
+                              .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean({EventStorageEngine.class, EventBus.class})
+    @ConditionalOnProperty(value = "axon.mongo.event-store.enabled")
```

Comment:
Don't you need to specify the `matchIfMissing` field for this bean creation method too?

Reply:
I'd rather not, since the default is false.

Reply:
But, `matchIfMissing` can equal `false`, right? Or do I not understand this parameter correctly?

Reply:
Yes, but that's the default, so that would not change anything.

Reply:
Aaah gotcha, wasn't aware. Then the only benefit is clarity, which isn't necessarily important for this class. Thanks for the explanation!